test/unit/text-field/Input.test.tsx

```diff
@@ -403,3 +405,38 @@ test('#inputElement should return the native input', () => {
   assert.equal(inputElement!.tagName.toLowerCase(), 'input');
   assert.isTrue(inputElement instanceof HTMLInputElement);
 });
+
+test('should call foundation.setValue() when foundation is new', () => {
+  const wrapper = shallow<Input<HTMLInputElement>>(<Input />);
+  const setValue = td.func();
+  const foundation = {setValue, setDisabled: td.func(), setUseNativeValidation: td.func()};
+  const handleValueChange = (_: any, callback: () => void) => callback();
+  const value = 'meow';
+  wrapper.setProps({foundation, value, handleValueChange});
+  td.verify(foundation.setValue(value), {times: 1});
+});
+
+test('shouldn\'t call foundation.setValue() when triggered by user', () => {
+  const setValue = td.func();
+  const foundation = {
+    setValue,
+    setDisabled: td.func(),
+    setUseNativeValidation: td.func(),
+    autoCompleteFocus: td.func(),
+  };
+  let value: any;
+  const handleValueChange = (v: any, callback: () => void) => {
+    value = v;
+    callback();
+  };
+  const wrapper = shallow<Input<HTMLInputElement>>(
+    <Input
+      handleValueChange={handleValueChange}
+      foundation={foundation}
+    />
+  );
+  const event = {target: {value: 'meow'}};
+  wrapper.simulate('change', event);
+  wrapper.setProps({value});
+  td.verify(foundation.setValue(), {times: 0});
```

Comment:
this would actually never be called. It needs to be called with foundation.setValue(value)